includes/alerts.inc.php

```diff
@@ -879,3 +884,25 @@ function ExtTransports($obj)
         echo '; ';
     }
 }//end ExtTransports()
+
+/**
+ * Check if a device's all parent are down
+ * Returns true if all parents are down
+ * @param int $device Device-ID
+ * @return bool
+ */
+function IsParentDown($device)
+{
+    $parent_count = dbFetchCell("SELECT count(`parent_device_id`) from `device_relationships` WHERE `child_device_id` = ?", array($device));
+    if (!$parent_id || $parent_id == null || $parent_id == 0) {
+        return false;
+    }
+
+
+    $down_parent_count = dbFetchCell("SELECT count(1) from devices as LEFT JOIN devices_attribs as a USING (device_id) LEFT JOIN device_relationships as r ON d.device_id=r.parent_device_id WHERE d.status=0 AND d.ignore=0 AND d.disabled=0 AND r.child_device_id=? AND (d.status_reason='icmp' OR (a.attrib_type='override_icmp_disable' AND a.attrib_value=true))", array($device));
```

Comment:
This query doesn't work for me.  This does:

```
SELECT count(1) from devices as d LEFT JOIN devices_attribs as a ON d.device_id=a.device_id LEFT JOIN device_relationships as r ON d.device_id=r.parent_device_id WHERE d.status=0 AND d.ignore=0 AND d.disabled=0 AND r.child_device_id=? AND (d.status_reason='icmp' OR (a.attrib_type='override_icmp_disable' AND a.attrib_value=true))
```

Reply:
`count(1)` ??? this also should be `COUNT(*)`